rig/machine_control/struct_file.py

```diff
@@ -0,0 +1,122 @@
+"""Read struct files for SARK/SC&MP.
+"""
+import collections
+import re
+
+
+def read_struct_file(fp):
+    """Read a struct file defining the location of variables in memory.
+
+    Parameters
+    ----------
+    fp : file
+        A file-object containing the struct file to read.
+    """
+    # Holders for all structs
+    structs = dict()
+
+    # Holders for the current struct
+    name = None
+
+    # Iterate over every line in the file
+    for i, l in enumerate(fp):
+        # Empty the line of comments, if the line is empty then skip to the
+        # next line.  Split on whitespace to get the tokens.
+        tokens = re_comment.sub(b"", l).strip().split()
```

Comment:
This is because of fun Python2/3 changes (default strings in Python 2 = bytes, in Python 3 = unicode)...  I'm open to suggestions!